### PR TITLE
class-based-sharding-spring-boot-jpa-example

### LINKED ISSUE #15713: Add `class-based-sharding-spring-boot-jpa-example` to `class-based-sharding-algorithm-example`
Hi community,

### Background

[class-based-sharding-algorithm-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example) is an example of a custom algorithm that demonstrates how to integrate a user-defined sharding algorithm and declare the configuration using `CLASS_BASED` (avoid using SPI extensions)

### Aim

Add `class-based-sharding-spring-boot-jpa-example` for `class-based-sharding-algorithm-example`

### Basic Qualifications

- Java
- Maven

### Example FYI

- [sharding-spring-boot-jpa-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/sharding-example/sharding-spring-boot-jpa-example)

### PR DESCRIPTION
**Pull request by Swastyy**
Fixes #15713 

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/extension/classbased/sharding/spring/boot/jpa/ClassBasedShardingSpringBootJpaExample.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@EntityScan(basePackages = "org.apache.shardingsphere.example.core.jpa.entity")
+@SpringBootApplication(exclude = JtaAutoConfiguration.class)
+public class ClassBasedShardingSpringBootJpaExample {
+    
+    public static void main(final String[] args) throws SQLException {
+        try (ConfigurableApplicationContext applicationContext = SpringApplication.run(ClassBasedShardingSpringBootJpaExample.class, args)) {
+            ExampleExecuteTemplate.run(applicationContext.getBean(ExampleService.class));
+        }
+    }
+}
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/extension/classbased/sharding/spring/boot/jpa/ClassBasedShardingSpringBootJpaExample.java:41]: Please keep a blank line at the end of file.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
Hi @Swastyy , 
This module is for `CLASS_BASED` sharding algorithm, please do not use `AUTO` sharding algorithm.

You can refer to `class-based-sharding-spring-boot-mybatis-example`.

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/extension/classbased/sharding/spring/boot/jpa/ClassBasedShardingSpringBootJpaExample.java (hunk 1 above)
Comment (Swastyy) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/extension/classbased/sharding/spring/boot/jpa/ClassBasedShardingSpringBootJpaExample.java:41]: done

**Comment by RaigorJiang** [comment]
@Swastyy 
The algorithm type should be `CLASS_BASED`, but there just `INLINE` for now.

Ref: https://github.com/apache/shardingsphere/blob/master/examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-mybatis-example/src/main/resources/application-sharding-tables.properties

**Comment by Swastyy** [comment]
> @Swastyy Can you run a test of this example? Because I found that the specified algorithm implementation class belongs to the `mybatis` module.
> 
> E.g: `org.apache.shardingsphere.example.extension.classbased.sharding.spring.boot.mybatis.fixture.ClassBasedDataSourceStandardShardingAlgorithmFixture`

Hi @RaigorJiang , after opening examples folder which command to run it particular?

File: examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/resources/application-sharding-tables.properties [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+spring.shardingsphere.rules.sharding.tables.t_order_item.key-generate-strategy.column=order_item_id
+spring.shardingsphere.rules.sharding.tables.t_order_item.key-generate-strategy.key-generator-name=snowflake
+
+spring.shardingsphere.rules.sharding.binding-tables[0]=t_order,t_order_item
+spring.shardingsphere.rules.sharding.broadcast-tables=t_address
+
+spring.shardingsphere.rules.sharding.sharding-algorithms.t-order-class-based.type=CLASS_BASED
+spring.shardingsphere.rules.sharding.sharding-algorithms.t-order-class-based.props.strategy=standard
+spring.shardingsphere.rules.sharding.sharding-algorithms.t-order-class-based.props.algorithmClassName=org.apache.shardingsphere.example.extension.classbased.sharding.spring.boot.jpa.fixture.ClassBasedOrderStandardShardingAlgorithmFixture
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/resources/application-sharding-tables.properties:45]: Hi @Swastyy 
The class may be not exist in this module?

Comment (Swastyy) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/resources/application-sharding-tables.properties:45]: yes, @RaigorJiang these 3 `ClassBasedDataSourceStandardShardingAlgorithmFixture`, `ClassBasedOrderStandardShardingAlgorithmFixture`, `ClassBasedOrderItemStandardShardingAlgorithmFixture` are I need to implement here.

Comment (RaigorJiang) [reply examples/shardingsphere-jdbc-example/single-feature-example/extension-example/custom-sharding-algortihm-example/class-based-sharding-algorithm-example/class-based-sharding-spring-boot-jpa-example/src/main/resources/application-sharding-tables.properties:45]: Hi @Swastyy , any update?

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
@Swastyy  Merged, thank you!

**Comment by Swastyy** [comment]
> @Swastyy Merged, thank you!

yeah, and sorry for completing it so late. 😃

**Comment by RaigorJiang** [comment]
> yeah, and sorry for completing it so late. 😃

Don't worry, It's OK.

**Pull request merged**